rd-net/RdFramework/Impl/SocketWire.cs

```diff
@@ -376,14 +376,21 @@ protected void AddTerminationActions([NotNull] Thread receiverThread)
               Monitor.PulseAll(Lock);
             }
 
+            CloseServerSocket();
```

Comment:
Server socket is already closed on server lifetime termination on line 502 (old) / 513 (new).
However, due to LIFO termination order of lifetimes, AddTerminationAction (and Thread.Join) are executed before socket-closing action from CreateServerSocket.
A possibly better approach would be to move all thread-joining actions later in termination order by adding the to a child LifetimeDef added to wire lifetime early in wire lifecycle. It's up for discussion though.

Reply:
Nice catch, thanks. Didn't expect to find CloseSocket there. Introducing a child lifetime seems like over engineering to me. What about a contract of ownership transfer when SocketWire is created? I updated the review.